src/storage/src/sink/kafka.rs

```diff
@@ -1235,6 +1218,12 @@ where
                         None => record,
                     };
 
+                    let ts_bytes = ts.to_string().into_bytes();
+                    let record = record.headers(OwnedHeaders::new().insert(Header {
+                        key: "mz-timestamp",
```

Comment:
OTOH [Confluent's blog uses dashes](https://www.confluent.io/blog/5-things-every-kafka-developer-should-know/#tip-5-record-headers). I love a good hyphen, so I'll leave it as-is unless I hear otherwise.

Reply:
I have no real opinion but maybe you can drop a note in `#devex` on slack and someone might chime in?

But maybe we can make this a constant?